api.py

```
@@ -143,21 +162,31 @@ def generate():
+        result = output_function(
+            {"question": question.text, "chat_history": []}, callbacks=[]
+        )
+    except Exception as e:
+        logger.error(f"Error processing query: {e}")
+        raise HTTPException(status_code=500, detail="Error processing query")
 
     return {"result": result["answer"], "model": llm_name}
 
 
 @app.get("/generate-ticket")
 async def generate_ticket_api(question: BaseTicket = Depends()):
```

Ruff (0.8.2)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)